### ISSUE TITLE
Missing stack settings in the portal for azurerm_linux_web_app

### ISSUE THREAD

**Issue by asarma2307** [hashicorp/terraform-provider-azurerm#22889]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.3

### AzureRM Provider Version

3.67

### Affected Resource(s)/Data Source(s)

azurerm_linux_web_app

### Terraform Configuration Files

```hcl
resource "azurerm_linux_web_app" "my_app" {
  name                = "my-linux-app"
  resource_group_name = azurerm_resource_group.dojo_rg.name
  location            = azurerm_resource_group.dojo_rg.location
  tags                = var.tags
  service_plan_id     = module.my_plan.id
  site_config         {
    ftps_state = "FtpsOnly"
    application_stack  {
      java_version             = "17"
      java_server = "JAVA"
      java_server_version = "SE"
    }
  }
}
```

### Debug Output/Panic Output

```shell
application_stack {
              + java_server              = "JAVA"
              + java_server_version      = "SE"
              + java_version             = "17"
            }
        }
```

### Expected Behaviour

Web app should be provisioned with the correct values being reflected in the stack settings in the portal. 

### Actual Behaviour

Stack settings are missing on provisioning of the resource. 
![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/57169230/ed20b5d5-a95b-4a71-877d-fdb051e19f95)
Furthermore, if we change the settings from the portal and save the changes, it is not reflected in the tfstate file i.e the settings are the same as what was initially set in terraform. 

### Steps to Reproduce

terraform init
terraform apply

**Comment by xiaxyi** [comment]
Thanks @asarma2307 for raising this issue, based on my understanding of your config, you are creating the linux web app on the latest java server version and the value for that in Azure portal is "SE", so you put "SE" in your config, please correct me if there is any misunderstanding. 

Have you tried below config and see if it works? 
```
  site_config {
    application_stack {
      java_version        = "17"
      java_server         = "JAVA"
      java_server_version = "17"
    }
  }
```

**Comment by asarma2307** [comment]
> Thanks @asarma2307 for raising this issue, based on my understanding of your config, you are creating the linux web app on the latest java server version and the value for that in Azure portal is "SE", so you put "SE" in your config, please correct me if there is any misunderstanding.
> 
> Have you tried below config and see if it works?
> 
> ```
>   site_config {
>     application_stack {
>       java_version        = "17"
>       java_server         = "JAVA"
>       java_server_version = "17"
>     }
>   }
> ```

Yes, that is what I am trying to do. I did try out the configuration mentioned but I'm still facing the same issue.

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against v3.x of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
